process_pr.py

```diff
@@ -1460,92 +1480,217 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
+                # )
+                # commit_statuses = old_commit_statuses
+                # bot_status = get_status(bot_status_name, commit_statuses)
+                # for status in old_commit_statuses:
+                #     import github
+                #
+                #     if not dryRun:
+                #         last_commit_obj.create_status(
+                #             state=status.state,
+                #             description=status.description,
+                #             target_url=status.target_url or github.GithubObject.NotSet,
+                #             context=status.context,
+                #         )
+                #
+                #     for pre_check in pre_checks + extra_pre_checks:
+                #         if "%s/%s" % (cms_status_prefix, pre_check) == status.context:
+                #             pre_checks_state[pre_check] = status.state
+                #             pre_checks_url[pre_check] = status.target_url
+                #             break
+                #
+                #     for pre_check in pre_checks + extra_pre_checks:
+                #         if pre_checks_state.get(pre_check) is None:
+                #             pre_checks_state[pre_check] = ""
+                #
+                # # Restore tests and code-checks labels. The logic to set the former is quite complicated, so we trust old labels
+                # for lab in old_labels:
+                #     if lab.startswith("code-checks"):
+                #         signatures["code-checks"] = lab.split("-", 2)[2]
+                #     if lab.startswith("tests"):
+                #         signatures["tests"] = lab.split("-", 1)[1]
+
+            # Mark removed commits in cache as squashed
+            for commit_sha in missing_commits:
+                print("Marked commit {0} as squashed".format(commit_sha))
+                bot_cache[commit_sha]["squashed"] = True
+
+        for commit in all_commits:
+            if commit.sha not in bot_cache["commits"]:
+                bot_cache["commits"][commit.sha] = {
+                    "time": int(commit.commit.committer.date.timestamp()),
+                    "squashed": False,
+                }
+                if len(commit.parents) > 1:
+                    bot_cache["commits"][commit.sha]["files"] = []
+                else:
+                    bot_cache["commits"][commit.sha]["files"] = sorted(
+                        x["filename"] for x in get_commit(repo.full_name, commit.sha)["files"]
+                    )
+
+            elif len(commit.parents) > 1:
+                bot_cache["commits"][commit.sha]["files"] = []
+
+            if (
+                last_seen_commit_time
+                and commit.commit.committer.date.timestamp() > last_seen_commit_time
+            ):
+                bot_cache["commits"][commit.sha]["files"] = []
+                bot_cache["commits"][commit.sha]["time"] = last_seen_commit_time
```

Comment:
@iarspider , are you still working on the part which we discussed about setting empty list of files for all commits which are in PR but not in cache if there are no changes between last seen and last pushed changes?